client/app/controllers/collection/add.js

```diff
@@ -194,16 +260,37 @@ export default Ember.Controller.extend({
 
     //},
     saveParameter_signature: ['parameter', 'updated_parameter'],
-    saveParameter: function(parameter, updated_parameter) {
-        if (typeof updated_parameter.value !== undefined) {
-            parameter.value = updated_parameter.value;
+    saveParameter: function(parameter, updated_parameter, parameters) {
+        if (typeof updated_parameter.value !== 'undefined') {
+            Ember.set(parameter, 'value', updated_parameter.value);
         }
-        if (typeof updated_parameter.state !== undefined) {
-            parameter.state = updated_parameter.state;
+        if (typeof updated_parameter.state !== 'undefined') {
+            Ember.set(parameter, 'state', updated_parameter.state);
         }
+        Ember.run.next(this, function() {
+            this.get('updateState').call(this, this.get('formActions'));
+        });
+    },
+
+
+    closeSection_signature: ['sectionName'],
+    closeSection: function(sectionName, parameters) {
+        this.get('panelActions').close(this.get(`_names.${this.get('_names').indexOf(sectionName)}`));
+        this.get('parameters')[this.get('sections').find(section => section.name === sectionName).param].state = ['closed', 'saved'];
         this.get('updateState').call(this, this.get('formActions'));
     },
 
+    openSection_signature: ['sectionName'],
+    openSection: function(sectionName, parameters) {
+        this.get('panelActions').open(this.get(`_names.${this.get('_names').indexOf(sectionName)}`));
+        this.get('parameters')[this.get('sections').find(section => section.name === sectionName).param].state = ['open', 'editing'];
+    },
+
+    browserAlert_signature: ['alertString'],
+    browserAlert: function(alertString) {
+        alert(alertString);
+    },
```

Comment:
When would a user see a browser alert? Ideally we would want to show users toast notifications on success/error, for instance, when a preprint file is uploaded: 

![screen shot 2017-06-05 at 2 32 28 pm](https://cloud.githubusercontent.com/assets/6414394/26797730/e20e4faa-49fb-11e7-9380-e26dc3f9fc38.png)